tests/link/jax/test_subtensor.py

```diff
@@ -13,54 +15,62 @@
 
 
 def test_jax_Subtensor_constant():
+    shape = (3, 4, 5)
+    x_pt = tensor("x", shape=shape, dtype="int")
+    x_np = np.arange(np.prod(shape)).reshape(shape)
+
     # Basic indices
-    x_pt = pt.as_tensor(np.arange(3 * 4 * 5).reshape((3, 4, 5)))
     out_pt = x_pt[1, 2, 0]
     assert isinstance(out_pt.owner.op, pt_subtensor.Subtensor)
-    out_fg = FunctionGraph([], [out_pt])
-    compare_jax_and_py(out_fg, [])
+    out_fg = FunctionGraph([x_pt], [out_pt])
+    compare_jax_and_py(out_fg, [x_np])
 
     out_pt = x_pt[1:, 1, :]
     assert isinstance(out_pt.owner.op, pt_subtensor.Subtensor)
-    out_fg = FunctionGraph([], [out_pt])
-    compare_jax_and_py(out_fg, [])
+    out_fg = FunctionGraph([x_pt], [out_pt])
+    compare_jax_and_py(out_fg, [x_np])
 
     out_pt = x_pt[:2, 1, :]
     assert isinstance(out_pt.owner.op, pt_subtensor.Subtensor)
-    out_fg = FunctionGraph([], [out_pt])
-    compare_jax_and_py(out_fg, [])
+    out_fg = FunctionGraph([x_pt], [out_pt])
+    compare_jax_and_py(out_fg, [x_np])
 
     out_pt = x_pt[1:2, 1, :]
     assert isinstance(out_pt.owner.op, pt_subtensor.Subtensor)
-    out_fg = FunctionGraph([], [out_pt])
-    compare_jax_and_py(out_fg, [])
+    out_fg = FunctionGraph([x_pt], [out_pt])
+    compare_jax_and_py(out_fg, [x_np])
 
     # Advanced indexing
     out_pt = pt_subtensor.advanced_subtensor1(x_pt, [1, 2])
     assert isinstance(out_pt.owner.op, pt_subtensor.AdvancedSubtensor1)
-    out_fg = FunctionGraph([], [out_pt])
-    compare_jax_and_py(out_fg, [])
+    out_fg = FunctionGraph([x_pt], [out_pt])
+    compare_jax_and_py(out_fg, [x_np])
 
     out_pt = x_pt[[1, 2], [2, 3]]
     assert isinstance(out_pt.owner.op, pt_subtensor.AdvancedSubtensor)
-    out_fg = FunctionGraph([], [out_pt])
-    compare_jax_and_py(out_fg, [])
+    out_fg = FunctionGraph([x_pt], [out_pt])
+    compare_jax_and_py(out_fg, [x_np])
 
     # Advanced and basic indexing
     out_pt = x_pt[[1, 2], :]
     assert isinstance(out_pt.owner.op, pt_subtensor.AdvancedSubtensor)
-    out_fg = FunctionGraph([], [out_pt])
-    compare_jax_and_py(out_fg, [])
+    out_fg = FunctionGraph([x_pt], [out_pt])
+    compare_jax_and_py(out_fg, [x_np])
 
     out_pt = x_pt[[1, 2], :, [3, 4]]
     assert isinstance(out_pt.owner.op, pt_subtensor.AdvancedSubtensor)
-    out_fg = FunctionGraph([], [out_pt])
-    compare_jax_and_py(out_fg, [])
+    out_fg = FunctionGraph([x_pt], [out_pt])
+    compare_jax_and_py(out_fg, [x_np])
 
     # Flipping
     out_pt = x_pt[::-1]
-    out_fg = FunctionGraph([], [out_pt])
-    compare_jax_and_py(out_fg, [])
+    out_fg = FunctionGraph([x_pt], [out_pt])
+    compare_jax_and_py(out_fg, [x_np])
+
+    # Boolean indexing should work if indexes are constant
+    out_pt = x_pt[np.random.binomial(1, 0.5, size=(3, 4, 5))]
```

Comment:
This was the case that led me to open this PR. JAX is happy to do it but our check wouldn't let it compile